amp-pwa-reader/src/js/Article.js

```diff
@@ -0,0 +1,250 @@
+    }
+
+    let offset = (innerWidth * this.card.imageData.ratio) / 2;
+    this.container.style.transform = 'translateY(' + (offset + scrollY) + 'px)';
+
+    return new Promise(resolve => {
+      return this.container.animate([
+        { opacity: 1, transform: 'translateY(' + (scrollY) + 'px)' },
+        { opacity: 0, transform: 'translateY(' + (offset + scrollY) + 'px)' }
+      ], { duration: this.getAnimationSpeed(), easing: 'ease-out' }).onfinish = resolve;
+    });
+
+  }
+
+  render() {
+
+    // Create an empty container for the AMP page
+    this.container = this.createShadowRoot();
+
+    // Tell Shadow AMP to initialize the AMP page in prerender-mode
+    this.ampDoc = AMP.attachShadowDoc(this.container, this.doc, this.url);
+    this.ampDoc.setVisibilityState('prerender');
+
+    return this.ampDoc.ampdoc.whenReady();
+
+  }
+
+  show(replaceHistoryState) {
+
+    // We need to clone the featured image into the Shadow DOM so it scrolls
+    // along. There are cases were we don't have a linked card from the list
+    // view (e.g. we load directly into the article), in which case we need to
+    // generate a new one.
+    var card = this.card ? this.cloneCard() : this.generateCard();
+    this.ampDoc.ampdoc.getBody().prepend(card);
+
+    return new Promise(resolve => {
+
+      // animate the article in. Only makes sense when there's a card transition
+      // at the same time, within animateIn, we check for the availability of a
+      // connected card, and don't animate if it's not around.
+      this.animateIn().then(() => {
+
+        // Hide the original card, show the cloned one (this also animates)
+        if (this.card) {
+          card.style.opacity = '1';
+        }
+
+        // add class to html element for to contain the scroll, and transform
+        // the hamburger into a 'back' button.
+        document.documentElement.classList.add('sr-article-shown');
+
+        this.takeoverScroll();
+
+        // Set the visibility state of the AMP doc to visible
+        this.ampDoc.setVisibilityState('visible');
+
+        // Finally, add new history entry
+        // Note: We're doing this deliberately late due to an AMP
+        // Bug that overrides the history state object early on
+        shadowReader.nav.setOpenArticle(this, replaceHistoryState);
+
+        resolve();
+      });
+    });
+
+  }
+
+  hide() {
+
+    // remove class to html element for global CSS stuff
+    document.documentElement.classList.remove('sr-article-shown');
+
+    // for some reason the browser restores scroll lazily, so we need
+    // to wait a few ms until scrollTop can be set again..
+    setTimeout(() => {
+
+      this.restoreScroll();
+
+      // Show the card header, hide the cloned one
+      if (this.card) {
+        this.clonedCard.style.opacity = '0';
+      }
+
+      // animate everything back to the card/listing view, then
+      // clear the old Shadow DOM to free up memory.
+      return new Promise(resolve => {
```

Comment:
Can this be simplified as:

    return this.animateOut().then(() => {
       this.clear();
       return true;   
    }